src/pipe.rs

```diff
@@ -43,6 +42,29 @@ use crate::{
     types::KeepaliveStatus,
 };
 
+enum AuthenticatorError {
+    ChannelBusy,
+    InvalidChannel,
+    InvalidCommand,
+    InvalidLength,
+    InvalidSeq,
+    Timeout,
+}
```

Comment:
Since not all error codes are in this enum, maybe add an indication that some are missing and can be added on a need-to-use basis (I suppose they're not there to avoid unused warnings?).

Reply:
I moved the link to the spec from the `From` impl to the struct and added the unused variants in a comment.